runtime/caml/misc.h

```diff
@@ -180,14 +180,24 @@ CAMLdeprecated_typedef(addr, char *);
   #define CAMLunused
 #endif
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 /* GC timing hooks. These can be assigned by the user. These hooks
    must not allocate, change any heap value, nor call OCaml code. They
    can obtain the domain id with Caml_state->id. These functions must
    be reentrant. */
+#ifdef __cplusplus
+
+extern "C++" {
```

Comment:
Why would `extern "C++"` be needed if the code is already guarded with `__cplusplus`? I don't think it's correct anyway.

Reply:
`camlatomic.h` does that, so it seemed sensible: https://github.com/ocaml/ocaml/blob/ce1a0a50266a9a0dc770967e12e076db055395c1/runtime/caml/camlatomic.h#L27-L39
I suppose that could be useful if the include is inside an `extern "C"` section. I'm not sure that this is common.